src/com/google/enterprise/adaptor/prebuilt/PrebuiltTransforms.java

```diff
@@ -335,48 +467,81 @@ public static MetadataTransform replaceMetadata(
   private static class ReplaceTransform implements MetadataTransform {
-    private final List<String> keys;
+    private final List<Key> keys;
     private final Pattern toMatch;
     private final String replacement;
     private final boolean overwrite;
 
-    public ReplaceTransform(Collection<String> keys, Pattern toMatch,
+    public ReplaceTransform(List<Key> keys, Pattern toMatch,
         String replacement, boolean overwrite) {
-      this.keys = Collections.unmodifiableList(new ArrayList<String>(keys));
+      this.keys = keys;
       this.toMatch = toMatch;
       this.replacement = replacement;
       this.overwrite = overwrite;
     }
 
     @Override
     public void transform(Metadata metadata, Map<String, String> params) {
-      for (String key : keys) {
-        Set<String> original = metadata.getAllValues(key);
-        if (original.isEmpty()) {
-          log.log(Level.FINE, "No values for {0}. Skipping", key);
-          continue;
+      for (Key key : keys) {
+        switch (key.keyset) {
+          case METADATA:
+            replaceInMetadata(key.key, metadata);
+            break;
+          case PARAMS:
+            replaceInParams(key.key, params);
+            break;
         }
-        log.log(Level.FINE, "Replacing values that match {0} with {1}: {2}",
-            new Object[] {toMatch, replacement, original});
-        Set<String> values = new HashSet<String>(original);
-        for (String value : original) {
-          String newValue = toMatch.matcher(value).replaceAll(replacement);
-          if (overwrite) {
-            values.remove(value);
-          }
-          values.add(newValue);
+      }
+    }
+
+    private void replaceInMetadata(String key, Metadata metadata) {
+      Set<String> original = metadata.getAllValues(key);
+      if (original.isEmpty()) {
+        log.log(Level.FINE, "No metadata values for {0}. Skipping", key);
+        return;
+      }
+      log.log(Level.FINE,
+          "Replacing metadata values of {0} that match {1} with {2}: {3}",
+          new Object[] {key, toMatch, replacement, original});
+      Set<String> values = new HashSet<String>(original);
+      for (String value : original) {
+        String newValue = toMatch.matcher(value).replaceAll(replacement);
+        if (overwrite) {
+          values.remove(value);
         }
-        log.log(Level.FINE, "After replacing: {0}", values);
-        metadata.set(key, values);
+        values.add(newValue);
+      }
+      metadata.set(key, values);
+      log.log(Level.FINE, "After replacing metadata values for {0}: {1}",
+          new Object [] {key, values});
+    }
+
+    private void replaceInParams(String key, Map<String, String> params) {
+      String original = getTrimmedValue(params, key);
+      if (original == null) {
+        log.log(Level.FINE, "No param value for {0}. Skipping", key);
+        return;
       }
+      log.log(Level.FINE,
+          "Replacing param value of {0} that match {1} with {2}: {3}",
```

Comment:
s/match/matches/, but this whole log message seems deceptive (along with its metadata predecessor). Something like "Replacing {1} with {2} in param value of {0}: {3}", and similarly for the metadata log message.